redisq/queue.go

```diff
@@ -203,13 +208,20 @@ func (q *Queue) createStreamGroup() {
 }
 
 func (q *Queue) Release(msg *taskq.Message) error {
-	err := q.redis.XDel(q.stream, msg.ID).Err()
+	// Make the delete and re-queue operation atomic in case we crash midway and lose a message
+	pipe := q.redis.TxPipeline()
+	err := pipe.XDel(q.stream, msg.ID).Err()
```

Comment:
Err() is always nil since Pipeline is not yet executed, but I guess it is fine to double check